pkg/apis/pingcap/v1alpha1/validation/validation.go

```diff
@@ -116,6 +120,21 @@ func validatePDSpec(spec *v1alpha1.PDSpec, fldPath *field.Path) field.ErrorList
 	return allErrs
 }
 
+func validatePDAddress(arrayOfAddresses []string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	addressRegex := "^http://[-A-Za-z0-9+&@#/%?=~_|!:,.;]+[-A-Za-z0-9+&@#/%=~_|]:[0-9]+$"
```

Comment:
Can we use `url.Parse` to simply the logical?

Reply:
This makes code easier to understand. I've modified it.